pkg/api/validation/validation.go

```diff
@@ -1353,14 +1371,18 @@ func ValidatePodSpec(spec *api.PodSpec, fldPath *field.Path) field.ErrorList {
 	allErrs = append(allErrs, ValidatePodSecurityContext(spec.SecurityContext, spec, fldPath, fldPath.Child("securityContext"))...)
 	allErrs = append(allErrs, validateImagePullSecrets(spec.ImagePullSecrets, fldPath.Child("imagePullSecrets"))...)
 	if len(spec.ServiceAccountName) > 0 {
-		if ok, msg := ValidateServiceAccountName(spec.ServiceAccountName, false); !ok {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("serviceAccountName"), spec.ServiceAccountName, msg))
+		if ok, msgs := ValidateServiceAccountName(spec.ServiceAccountName, false); !ok {
+			for i := range msgs {
+				allErrs = append(allErrs, field.Invalid(fldPath.Child("serviceAccountName"), spec.ServiceAccountName, msgs[i]))
+			}
 		}
 	}
 
 	if len(spec.NodeName) > 0 {
-		if ok, msg := ValidateNodeName(spec.NodeName, false); !ok {
-			allErrs = append(allErrs, field.Invalid(fldPath.Child("nodeName"), spec.NodeName, msg))
+		if ok, msgs := ValidateNodeName(spec.NodeName, false); !ok {
+			for i := range msgs {
```

Comment:
Yeah, if you got rid of the bool, you could change all of these loops to be

```go
for _, msg := range ValidateNodeName(spec.NodeName, false) {
  append(...)
}
```

which would remove a level of indentation and be a little easier to parse.